lib/Support/Check.cpp

```diff
@@ -13,12 +13,34 @@
 #include "llvm/Support/Error.h"
 #include "llvm/Support/raw_ostream.h"
 
-static bool isDenorm(float F) { return std::fpclassify(F) == FP_SUBNORMAL; }
+template <typename T> static bool isDenorm(T F) {
+  return std::fpclassify(F) == FP_SUBNORMAL;
+}
 
 static bool isFloat16NAN(uint16_t Val) {
   return (Val & 0x7c00) == 0x7c00 && (Val & 0x03ff) != 0;
 }
 
+static bool compareDoubleULP(const double &FSrc, const double &FRef,
+                             unsigned ULPTolerance,
```

Comment:
Should `ULPTolerance` be a `uint64_t`?
Its use is to be compared to `AbsDiff` which is a `const uint64_t`